drivers/sensor/tmp116/tmp116.c

```diff
@@ -112,39 +122,78 @@ static int tmp116_channel_get(const struct device *dev,
 	return 0;
 }
 
+static int tmp116_attr_set(const struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	int16_t value, id;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_OFFSET:
+		tmp116_device_id_check(dev, &id);
+		if (id != TMP117_DEVICE_ID) {
+			LOG_ERR("%s: Offset is only supported by TMP117",
+			dev->name);
+			return -EINVAL;
+		}
+		/*
+		 * The offset is encoded into the temperature register format.
+		 */
+		value = (((val->val1) * 10000000) + ((val->val2) * 1000000))
+						/ (int32_t)TMP116_RESOLUTION;
+
+		return tmp116_reg_write(dev, TMP117_REG_TEMP_OFFSET, value);
+
+	default:
+		return -ENOTSUP;
+	}
+}
+
 static const struct sensor_driver_api tmp116_driver_api = {
+	.attr_set = tmp116_attr_set,
 	.sample_fetch = tmp116_sample_fetch,
 	.channel_get = tmp116_channel_get
 };
 
 static int tmp116_init(const struct device *dev)
 {
 	struct tmp116_data *drv_data = dev->data;
+	const struct tmp116_dev_config *cfg = dev->config;
 	int rc;
+	uint16_t id;
 
 	/* Bind to the I2C bus that the sensor is connected */
-	drv_data->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
+	drv_data->i2c = device_get_binding(cfg->i2c_bus_label);
 	if (!drv_data->i2c) {
 		LOG_ERR("Cannot bind to %s device!",
-			DT_INST_BUS_LABEL(0));
+			cfg->i2c_bus_label);
 		return -EINVAL;
 	}
 
 	/* Check the Device ID */
-	rc = tmp116_device_id_check(dev);
+	rc = tmp116_device_id_check(dev, &id);
 	if (rc < 0) {
 		return rc;
+	} else {
+		LOG_INF("Got device ID: %x", id);
```

Comment:
better use `LOG_DBG`, `else` branch not needed.

Reply:
done